components/js/SlickGrid.ts

```diff
@@ -435,8 +442,13 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
                     this.updateColumnWidths();
                 });
         }
+
         for (let plugin of this.plugins) {
-            this.registerPlugin(plugin);
+            if (typeof plugin === 'string') {
```

Comment:
You don't need this if check anymore, correct? `registerPlugin` accepts the plugin whether it's a string or a plugin object, so you can get rid of this?

Reply:
I think its a bug in typescript, if I don't add a type check if complains `string | object` not assignable to string.

Reply:
I guess I could fix it by not overloading the function and instead just having the input as string | object

Reply:
Looks like its a bug that they couldn't find a good way to fix, so they said just workaround it via my solution about just using a union type instead of an overload Microsoft/TypeScript#1805